src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+package mate.academy.dao.impl;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    private static final String URL = "jdbc:mysql://localhost:3306/your_schema?serverTimezone=UTC";
+    private static final String USER = "your_username";
+    private static final String PASSWORD = "your_password";
+
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.executeUpdate();
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to create book: " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(mapResultSetToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to find book by id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT * FROM books";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                Statement statement = connection.createStatement();
```

Comment:
Use prepared statement whenever it possible
```suggestion
                PrepatedStatement statement = connection.createStatement();
```